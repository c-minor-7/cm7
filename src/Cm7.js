import Song from './models/Song';

export default ({ el, cssClasses, ast }) => {
  // pre: all arguments are checked!
  const song = Song.fromAST(ast);
  const $el = document.querySelector(el);

  // remove all children in $el
  while ($el.firstChild) $el.removeChild($el.firstChild);

  $el.appendChild(song.toDOM({ cssClasses }));
};
