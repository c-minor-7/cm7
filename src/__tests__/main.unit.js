import Cm7 from '../main';

describe('cm7', () => {
  const src = `key=C

1 1M7/3
(L)ondon bridge is (f)alling down
2m 4madd4/7b
(F)alling (d)own
1
(F)alling down

1
(L)ondon bridge is falling down
2m
(F)alling down
1
(F)alling down`;

  afterEach(() => {
    document.body.innerHTML = '';
  });

  it('should replace the content of the target element that match snapshot', async () => {
    const id = 'my-chord-displayer';
    const content = 'original content';

    document.body.innerHTML = `<div id="${id}">${content}</div>`;

    Cm7({
      el: `#${id}`,
      src,
    });

    const innerHTML = document.querySelector(`#${id}`).innerHTML;
    expect(innerHTML).not.toBe(content);
    expect(innerHTML).toMatchSnapshot();
  });
});
