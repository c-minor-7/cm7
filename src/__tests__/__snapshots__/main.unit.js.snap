// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cm7 should replace the content of the target element that match snapshot 1`] = `"<div class=\\"cm7_song\\"><div class=\\"cm7_section\\"><div class=\\"cm7_line\\"><div class=\\"cm7_line_chords\\"><span class=\\"cm7_chord\\">C</span><span class=\\"cm7_chord\\">CM7/E</span></div><div class=\\"cm7_line_lyrics\\"><span class=\\"cm7_line_lyrics-beat\\">L</span>ondon bridge is <span class=\\"cm7_line_lyrics-beat\\">f</span>alling down</div></div><div class=\\"cm7_line\\"><div class=\\"cm7_line_chords\\"><span class=\\"cm7_chord\\">Dm</span><span class=\\"cm7_chord\\">Fmadd4/Bb</span></div><div class=\\"cm7_line_lyrics\\"><span class=\\"cm7_line_lyrics-beat\\">F</span>alling <span class=\\"cm7_line_lyrics-beat\\">d</span>own</div></div><div class=\\"cm7_line\\"><div class=\\"cm7_line_chords\\"><span class=\\"cm7_chord\\">C</span></div><div class=\\"cm7_line_lyrics\\"><span class=\\"cm7_line_lyrics-beat\\">F</span>alling down</div></div></div><div class=\\"cm7_section\\"><div class=\\"cm7_line\\"><div class=\\"cm7_line_chords\\"><span class=\\"cm7_chord\\">C</span></div><div class=\\"cm7_line_lyrics\\"><span class=\\"cm7_line_lyrics-beat\\">L</span>ondon bridge is falling down</div></div><div class=\\"cm7_line\\"><div class=\\"cm7_line_chords\\"><span class=\\"cm7_chord\\">Dm</span></div><div class=\\"cm7_line_lyrics\\"><span class=\\"cm7_line_lyrics-beat\\">F</span>alling down</div></div><div class=\\"cm7_line\\"><div class=\\"cm7_line_chords\\"><span class=\\"cm7_chord\\">C</span></div><div class=\\"cm7_line_lyrics\\"><span class=\\"cm7_line_lyrics-beat\\">F</span>alling down</div></div></div></div>"`;
